internal/controllers/port/controller.go

```diff
@@ -43,7 +43,15 @@ var (
 	networkDependency = dependency.NewDeletionGuardDependency[*orcv1alpha1.PortList, *orcv1alpha1.Network](
 		"spec.resource.networkRef",
 		func(port *orcv1alpha1.Port) []string {
-			return []string{string(port.Spec.NetworkRef)}
+			resource := port.Spec.Resource
+			if resource != nil {
+				return []string{string(resource.NetworkRef)}
+			}
+			importStruct := port.Spec.Import
+			if importStruct != nil && importStruct.Filter != nil {
+				return []string{string(importStruct.Filter.NetworkRef)}
+			}
```

Comment:
I think this is more complicated than this. This only needs a DeletionGuard dependency for the ResourceSpec. We might need 2 different dependencies on Network. This needs more thought.